### DISCUSSION TITLE
Investigate the possibility of testing elements

### DISCUSSION THREAD

**Discussion by djowel** [cycfi/elements#351] [Other features]
Testing GUIs is tricky. but nevertheless, I don't feel good without tests. Right now, I have to build the examples and inspect each one of them to make sure everything is good.

At the very least, a quick testing facility is probably something like:
1. Saving the view to a PNG
2. Comparing the PNGs to known 'golden' files and reporting the diffs.

It might be possible to simulate user interaction, but that can probably be done later.

**Comment by Xeverous** [comment]
You might want to expose an interface that allows issuing events programatically. Eg issue a virtual click at some coordinates and verify that state changed accordingly. If rendering in well-separated from the logic it should be trivial to test correct behavior.
